src/main/java/org/folio/consortia/security/SecurityManagerService.java

```diff
@@ -0,0 +1,117 @@
+
+import org.apache.commons.lang3.StringUtils;
+import org.folio.consortia.client.PermissionsClient;
+import org.folio.consortia.client.UsersClient;
+import org.folio.consortia.domain.dto.PermissionUser;
+import org.folio.consortia.domain.dto.Personal;
+import org.folio.consortia.domain.dto.SystemUserParameters;
+import org.folio.consortia.domain.dto.User;
+import org.folio.consortia.service.PermissionUserService;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Component
+@Log4j2
+@RequiredArgsConstructor
+public class SecurityManagerService {
+
+  private static final String PERMISSIONS_FILE_PATH = "permissions/system-user-permissions.csv";
+  private static final String USER_LAST_NAME = "SystemConsortia";
+
+  private final PermissionsClient permissionsClient;
+  private final UsersClient usersClient;
+  private final AuthService authService;
+  private final PermissionUserService permissionUserService;
+
+  @Value("${folio.system.username}")
+  private String username;
+
+  @Value("${folio.system.password}")
+  private String password;
+
+  public void prepareSystemUser(String okapiUrl, String tenantId) {
+    Optional<User> userOptional = getUser(username);
+
+    User user;
+    if (userOptional.isPresent()) {
+      user = userOptional.get();
+      updateUser(user);
+    } else {
+      user = createUser(username);
+      authService.saveCredentials(SystemUserParameters.builder()
+        .id(UUID.randomUUID())
+        .username(username)
+        .password(password)
+        .okapiUrl(okapiUrl)
+        .tenant(tenantId)
+        .build());
+    }
+
+    Optional<PermissionUser> permissionUserOptional = permissionsClient.get("userId==" + user.getId())
+      .getPermissionUsers()
+      .stream()
+      .findFirst();
+    if (permissionUserOptional.isPresent()) {
+      permissionUserService.addPermissions(permissionUserOptional.get(), PERMISSIONS_FILE_PATH);
+    } else {
+      permissionUserService.createWithPermissionsFromFile(user.getId(), PERMISSIONS_FILE_PATH);
+    }
+  }
+
+  private Optional<User> getUser(String username) {
+    return usersClient.getUsersByQuery("username==" + username)
+      .getUsers()
+      .stream()
+      .findFirst();
+  }
+
+  private User createUser(String username) {
+    var result = createUserObject(username);
+    log.info("Creating {}.", result);
+    usersClient.saveUser(result);
+    return result;
+  }
+
+  private void updateUser(User user) {
+    if (existingUserUpToDate(user)) {
+      log.info("{} is up to date.", user);
+    } else {
+      populateMissingUserProperties(user);
+      log.info("Updating {}.", user);
+      usersClient.updateUser(user.getId(), user);
+    }
+  }
+
+  private User createUserObject(String username) {
+    final var result = new User();
+
+    result.setId(UUID.randomUUID()
+      .toString());
```

Comment:
```suggestion
     result.setId(UUID.randomUUID().toString());
```
and the same for setting last name in populateMissingUserProperties()

Reply:
changed